docs/sphinx/user_guide/feature/sort.rst

```diff
@@ -0,0 +1,141 @@
+            ``CUB_DIR`` will be automatically set to the location of the cub
+            library when CUDA is enabled; to use a different version of the
+            cub library, install it and set the ``CUB_DIR`` variable to the
+            desired location when running CMake.
+          * The RAJA CUDA back-end implementation only supports sorting
+            arithmetic types using RAJA operators less and greater.
+
+Please see the :ref:`sort-label` tutorial section for usage examples of RAJA
+sort operations.
+
+-----------------
+Sort Operations
+-----------------
+
+In general, a sort operation takes a sequence of numbers 'x' and a binary
+comparison operator 'op' that forms a strict weak ordering of the elements in x
+as input and produces a sequence of numbers 'y' as output. The output sequence
+is a permutation of the input sequence where each pair of elements 'a' and 'b,'
+where a is before b in the output sequence, satisfies !(b op a).
+Sorts are stable if they always preserve the order of equivalent elements,
+where equivalent elements satisfy !(a op b) && !(b op a).
+
+A **stable sort** takes an input sequence
+
+   x = { a\ :sub:`0`\, b\ :sub:`0`\, a\ :sub:`1`\, ... }
+
+and calculates the output sequence:
+
+   y = { a\ :sub:`0`\, a\ :sub:`1`\, b\ :sub:`0`\, ... }
+
+An **unstable sort** may not preserve the order of equivalent elements and
+may produce either of the following output sequences.
+
+   y = { a\ :sub:`0`\, a\ :sub:`1`\, b\ :sub:`0`\, ... }
+
+   or
+
+   y = { a\ :sub:`1`\, a\ :sub:`0`\, b\ :sub:`0`\, ... }
+
+---------------------
+RAJA Unstable Sorts
+---------------------
+
+RAJA unstable sort operations look like the following:
+
+ * ``RAJA::sort< exec_policy >(container)``
+ * ``RAJA::sort< exec_policy >(container, comparator)``
+ * ``RAJA::sort< exec_policy >(iter, iter + N)``
+ * ``RAJA::sort< exec_policy >(iter, iter + N, comparator)``
+
+Here, 'container' is a range of elements and 'iter' is a random access
+iterator to a range of elements. 'container' and 'iter' provide access to the
+input sequence and contain the output sequence at the end of sort. The first
+and third sort operations above will be *non-decreasing* sorts since there is
+no comparator argument given; i.e., the sequences will be reordered *in-place*
+using operator::less. The second and fourth sorts will apply the comparator
+that is passed into the function.
+
+RAJA also provides sort pairs that operate on key, value pairs stored
+separately:
+
+ * ``RAJA::sort_pairs< exec_policy >(keys_container, vals_container)``
+ * ``RAJA::sort_pairs< exec_policy >(keys_container, vals_container, comparator)``
+ * ``RAJA::sort_pairs< exec_policy >(keys_iter, keys_iter + N, vals_iter)``
+ * ``RAJA::sort_pairs< exec_policy >(keys_iter, keys_iter + N, vals_iter, comparator)``
+
+Sort pairs generates the same output sequence of keys as sort and also reorders
+vals by permuting it the same as it did the keys; i.e. sorting the pairs based
```

Comment:
Add single quotes around vals to make it clearer?

Reply:
I kinda disagree here, double backticks to make it verbatim would make it clearer that it's a literal variable name, but only if the names actually match.  Currently, what are "keys" and "vals"? Are they the same as the values covered by `keys_iter` and `vals_iter`?

Reply:
@trws that's what I meant, but I was imprecise.  @MrBurmark you could do `vals_*` and `keys_*` to satisfy Tom's point.

Reply:
Sorry for the misunderstanding, thanks @rhornung67.

Reply:
@trws no worries. I shouldn't be commenting on PRs when I'm on a webex.